src/cryptonote_basic/miner.cpp

```diff
@@ -575,7 +577,26 @@ namespace cryptonote
         continue; // if interrupted because stop called, loop should end ..
       }
 
-      bool on_ac_power = !on_battery_power();
+      boost::tribool battery_powered(on_battery_power());
+      if(indeterminate( battery_powered ))
+      {
+        if( ++battery_query_failure_count > 3 )
+        {
+          MERROR("on_battery_power() call failed 3 times, shutting down mining");
+          stop();
+          break;
+        }
+        else
+        {
+          continue; // try again ..
```

Comment:
I'm not super familiar with these system files, or what errors could occur with opening, so I figured a few tries couldn't hurt just in case of some transient error?